src/components/PostDetails.tsx

```diff
@@ -1,106 +1,121 @@
+  addComment: (newComment: CommentData) => void;
+  buttonAddLoading: boolean;
+  setShowCommentForm: (item: boolean) => void;
+  showCommentForm: boolean;
+  userSelected: User | null;
+};
+
+export const PostDetails: React.FC<Props> = ({
+  showCommentLoader,
+  postComments,
+  selectedPost,
+  errorsComment,
+  handleDeleteComment,
+  addComment,
+  buttonAddLoading,
+  setShowCommentForm,
+  showCommentForm,
+  userSelected,
+}) => {
+  useEffect(() => {
+    setShowCommentForm(false);
+  }, [selectedPost]);
+
+  if (!selectedPost || userSelected?.id !== selectedPost.userId) {
+    return null;
+  }
 
-export const PostDetails: React.FC = () => {
   return (
     <div className="content" data-cy="PostDetails">
       <div className="content" data-cy="PostDetails">
         <div className="block">
-          <h2 data-cy="PostTitle">
-            #18: voluptate et itaque vero tempora molestiae
-          </h2>
+          <h2 data-cy="PostTitle">{`#${selectedPost?.id}: ${selectedPost?.title}`}</h2>
 
-          <p data-cy="PostBody">
-            eveniet quo quis laborum totam consequatur non dolor ut et est
-            repudiandae est voluptatem vel debitis et magnam
-          </p>
+          <p data-cy="PostBody">{selectedPost?.body}</p>
         </div>
 
         <div className="block">
-          <Loader />
-
-          <div className="notification is-danger" data-cy="CommentsError">
-            Something went wrong
-          </div>
-
-          <p className="title is-4" data-cy="NoCommentsMessage">
-            No comments yet
-          </p>
-
-          <p className="title is-4">Comments:</p>
+          {showCommentLoader ? (
+            <Loader />
+          ) : (
+            <>
+              {errorsComment ? (
+                <div className="notification is-danger" data-cy="CommentsError">
+                  Something went wrong
+                </div>
+              ) : (
+                <>
+                  {postComments.length === 0 ? (
+                    <p className="title is-4" data-cy="NoCommentsMessage">
+                      No comments yet
+                    </p>
+                  ) : (
+                    <p className="title is-4">Comments:</p>
+                  )}
 
-          <article className="message is-small" data-cy="Comment">
-            <div className="message-header">
-              <a href="mailto:misha@mate.academy" data-cy="CommentAuthor">
-                Misha Hrynko
-              </a>
-              <button
-                data-cy="CommentDelete"
-                type="button"
-                className="delete is-small"
-                aria-label="delete"
-              >
-                delete button
-              </button>
-            </div>
+                  {postComments.map((comment: Comment) => (
+                    <article
+                      className="message is-small"
+                      data-cy="Comment"
+                      key={comment.id}
```

Comment:
You should avoid generating keys on render. It's better to use unique and consistent values from your data, like `comment.id`. More details can be found in this [article](https://medium.com/blackrock-engineering/5-common-mistakes-with-keys-in-react-b86e82020052).